torchinductor/ir.py

```diff
@@ -1994,18 +2027,20 @@ def create(
                 V.graph.sizevars.guard_static_shape(output_size[-1])
             )
 
-        # memory layout order of output depends on x
-        size_hint = V.graph.sizevars.size_hint
-        x_stride = [size_hint(i) for i in x.get_stride()]
-        order = sorted(range(len(x_stride)), key=x_stride.__getitem__)
+        if any(k != 1 for k in output_size[-len(stride) :]) and in_channels_stride == 1:
+            # channels last format
+            order = [0] + list(reversed(range(1, len(kernel_size) + 1)))
+            if len(order) < len(output_size):
+                # add batch dim if it exists
+                order = [len(order)] + order
```

Comment:
Could we get fill order directly for channel last format
```
order = [1] + list(reversed(range(2, len(kernel_size)+2))
if len(order) < len(output_size):
  # add batch dim if it exists
  order = order + [0]
```
The `FlexbileLayout.fill_ordered(order)`?
Do we need `FlexibleLayout.stride_ordered()`?

Reply:
Yeah, we could.  IMO this is a bit more readable, though I suppose it is preference.